src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -887,4 +890,37 @@ default boolean shouldDisplayFluidOverlay(BlockState state, IBlockDisplayReader
     {
         return state.getBlock() instanceof BreakableBlock || state.getBlock() instanceof LeavesBlock;
     }
+    
+    /**
+     * Returns the state that this block should transform into when right clicked by a tool.
+     * Used to determine if an axe can strip, a shovel can path, or a hoe can till.
+     * Return null if vanilla behavior should be disabled.
+     *
+     * @param state The current state
+     * @param world The world
+     * @param pos The block position in world
+     * @param player The player clicking the block
+     * @param stack The stack being used by the player
+     * @return The resulting state after the action has been performed
+     */
+    @Nullable
+    default BlockState getToolModifiedState(BlockState state, World world, BlockPos pos, PlayerEntity player, ItemStack stack)
+    {
+    	Set<ToolType> tool = stack.getToolTypes();
```

Comment:
I'd say fire the event here, at the beginning and if it returns not the original state, then return that state.
Otherwise continue.

Reply:
I've added the event.